src/main/java/com/smartagilify/core/util/HashUtil.java

```diff
@@ -0,0 +1,50 @@
+package com.smartagilify.core.util;
+
+import com.smartagilify.core.constant.ErrorConstant;
+import com.smartagilify.core.exceptions.InternalServerException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+@Slf4j
+public class HashUtil {
+
+    private static final ThreadLocal<MessageDigest> messageDigestThreadLocal = ThreadLocal.withInitial(() -> {
```

Comment:
could you explain what happens here?

Reply:
I want to have SHA Message digest in my class and not construct it for every call. so i define it in static block.
then it could cause synchronize problems and other problems that create by a thread (because it's not thread-safe) so I define it in the thread local block.